File: 코딩테스트 대비/14719.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
   int h, w;
   cin >> h >> w;
   int answer = 0;
   vector<int> v(w);
   for (int i = 0; i < w; ++i)
      cin >> v[i];

   for (int i = 0; i < w; ++i)
   {
      int s = 0;
      for (int j = v[i]; j >= 0; --j)
         for (int k = i + 1; k < w and j > v[k]; ++k)
         {
            ++s;
            if (k == w - 1)
               s = 0;
         }
      answer += s;
   }
   cout << answer;

   return 0;
}

File: 코딩테스트 대비/14719.py
n, m = map(int, input().split())
arr = list(map(int, input().split()))
s = [arr[0]]
answer = 0
for i in range(1, m):
    if s[0] > arr[i]:
        s.append(arr[i])
    else:
        while len(s) > 0:
            answer += s[0] - s[-1]
            s.pop()
        s.append(arr[i])

arr = s[-1::-1]
s = [arr[0]]
for i in range(1, len(arr)):
    if s[0] > arr[i]:
        s.append(arr[i])
    else:
        while len(s) > 0:
            answer += s[0] - s[-1]
            s.pop()
        s.append(arr[i])

print(answer)